dev/ci/README-CI-Pipeline.md

```diff
@@ -0,0 +1,190 @@
+# Global Workflow CI/CD Pipeline Architecture
+
+This document describes the GitLab CI/CD pipeline architecture for the global-workflow project, including the different testing modalities, configuration methods, and operational details.
+
+## Overview
+
+The CI/CD system supports multiple testing approaches across different computing platforms, with flexible triggering mechanisms from both GitHub Actions and GitLab scheduled pipelines. The architecture is designed to be easily extensible to new computing hosts and testing scenarios.
+
+## Pipeline Modalities
+
+The pipeline supports two primary testing modalities controlled by the `PIPELINE_TYPE` variable:
+
+### 1. PR Cases (`pr_cases`)
+- **Purpose**: Comprehensive end-to-end testing of experiment workflows
+- **Default**: This is the default modality when no override is specified
+- **Test Scope**: Full experiment cases that validate complete workflow functionality
+- **Duration**: Longer running tests (typically hours)
+- **Use Cases**: 
+  - Pull request validation
+  - Nightly regression testing
+  - Full system integration testing
+
+### 2. CTests (`ctests`)
+- **Purpose**: Fast, focused testing of individual Rocoto jobs
+- **Test Scope**: Unit-level tests with predefined input data
+- **Duration**: Shorter running tests (typically minutes to an hour)
+- **Use Cases**:
+  - Quick PR validation
+  - Developer feedback during development
+  - Targeted testing of specific components
+
+## Configuration Methods
+
+### GitLab Scheduled Pipelines
+
+GitLab scheduled pipelines can override pipeline behavior using environment variable settings:
+
+#### Setting Pipeline Type
+```bash
+# In GitLab Project > CI/CD > Pipelines > Schedules
+# Set variables in the pipeline schedule configuration:
+
+PIPELINE_TYPE=ctests          # Override to run CTests
+PIPELINE_TYPE=pr_cases        # Explicit setting for PR cases (default behavior)
+```
```

Comment:
It's not clear to me where one would need to set these variables.  Is this something that would be done in GitLab?

If so, I think this shouldn't be `bash` markup but instead a numbered/bulleted list.  Also, I think the step is `In GitLab Project > Pipeline schedules`, though I cannot see these options (likely because I don't have GitLab pro).

![image](https://github.com/user-attachments/assets/4561f5e3-2ea1-45e9-8d6c-7d302f21f0cf)

Reply:
Oh yeah these are BASH values, I'll update the description to help with that.